### PR TITLE
Fppsp 234 - Upload a family image.

### PR DESCRIPTION
**Pull request by srpablino**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

Put an `x` in the boxes that apply.

* [x] Lint and unit tests pass locally with my changes
* [ ] I have added necessary documentation (if appropriate)
* [ ] I have added tests that prove my fix is effective or that my feature works (if appropiate)

### Description

A new endpoint PUT /api/v1/families/{family_id}/image is added, in order to allow the upload of an image for families. 

This adds the attribute image_url to both, FamilyEntity and FamilyDTO. So, now when families objects are queried from the clients, the URL of the image is accessible in the new attribute.

💔Thank you!

### CODE REVIEW

File: src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java [hunk 1]
```diff
@@ -126,6 +140,37 @@ public FamilyDTO updateFamily(Long familyId) {
                         new UnknownResourceException(i18n.translate("family.notExist")));
     }
 
+    @Override
+    public String imageUpload(Long idFamily, MultipartFile multipartFile) throws IOException {
+
+        FamilyEntity familyEntity= familyRepository.findOne(idFamily);
+
+        if (familyEntity==null){
+            throw new UnknownResourceException("Family does not exist");
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: Usar si es que hay un key de i18n en los message.properties.

File: src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java [hunk 2]
```diff
@@ -126,6 +140,37 @@ public FamilyDTO updateFamily(Long familyId) {
...
+        }
+
+        File file = File.createTempFile("file", ".tmp");
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: Utilizar el patrón try-with-resources: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html para poder liberar correctamente el recurso de outpustream. O bien utilizar alguno de los utilitarios de escritura de archivos de Guava o Apache Commons Lang.

File: src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java [hunk 3]
```diff
@@ -146,13 +146,14 @@ public String imageUpload(Long idFamily, MultipartFile multipartFile) throws IOE
...
-        FileOutputStream fos = new FileOutputStream(file);
-        fos.write(multipartFile.getBytes());
-        fos.close();
+
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+            fos.write(multipartFile.getBytes());
+        }
 
         String fileFormat = multipartFile.getContentType();
         ImageDTO image  = new ImageDTO();
```
Comment (rodrigojv) [inline src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: @srpablino existe una clase `ImageParser` que implementa una lógica similar, teniendo un método parse que te devuelve `ImageDTO`. Existe la posibilidad de reutilizar eso en este lugar para generar el ImageDTO?

Comment (srpablino) [reply src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: The first argument that ImageParser takes is a String.

I receive a multipartFile from the client, and could not find a way to convert it to the exactly string structure the ImageParser would handle. Besides, the ImageParser involves some other extra operations, which I don't know if are necessaries in this case.  

So, I decided to re-do a tiny part of the logic within the new service, to simplify the handle of the multipartFile input, and call directly the imageUploadService.

Comment (rodrigojv) [reply src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: Bien, podriamos entonces poner la lógica en otro método de ImageParser? Para delegar a esa clase la responsabilidad de creación de ImageDTO con distintos argumentos.

Comment (srpablino) [reply src/main/java/py/org/fundacionparaguaya/pspserver/families/services/impl/FamilyServiceImpl.java]: OK! No problem!

**Pull request merged**